recipes/openmathreasoning/scripts/merge_new_summary.py

```diff
@@ -62,20 +63,21 @@ def select_best_summary(valid_summaries):
 def trim_reasoning_generation(reasoning_generation, start_tag, end_tag, strict_end_tag=False):    
     """Trim the thinking part of the original reasoning generation till the step with the rightmost boxed entry"""
     
-    # Find the start and end tags. If either is not found, return None
+    # Find the start and end tags. If either is not found, use the entire generation
     start_tag_position = reasoning_generation.find(start_tag)
     if start_tag_position == -1:
-        return None
-
-    end_tag_position = reasoning_generation.find(end_tag)
-    if end_tag_position == -1:
-        if strict_end_tag:
-            return None
-        else:
-            reasoning_generation = reasoning_generation + end_tag
-            reasoning_trace = reasoning_generation
+        # If no start tag found, wrap the entire generation with tags
+        reasoning_trace = start_tag + "\n" + reasoning_generation + "\n" + end_tag
     else:
-        reasoning_trace = reasoning_generation[:end_tag_position + len(end_tag)]
+        end_tag_position = reasoning_generation.find(end_tag)
+        if end_tag_position == -1:
+            if strict_end_tag:
+                return None
+            else:
+                reasoning_generation = reasoning_generation + end_tag
```

Comment:
It would be good to follow a consistent format while adding these tags. A few lines up, a newline is added along with the tag. Would be good to keep the tag addition format consistent.